source/MaterialXTest/MaterialXCore/Node.cpp

```diff
@@ -627,97 +627,116 @@ TEST_CASE("Node Definition Creation", "[nodedef]")
-                    if (!valueElem)
-                    {
-                        continue;
-                    }
-                    REQUIRE(valueElem->getAttribute(mx::ValueElement::TYPE_ATTRIBUTE) == 
-                            nodeDefValueElem->getAttribute(mx::ValueElement::TYPE_ATTRIBUTE));
-                    REQUIRE(valueElem->getAttribute(mx::ValueElement::VALUE_ATTRIBUTE) == 
-                            nodeDefValueElem->getAttribute(mx::ValueElement::VALUE_ATTRIBUTE));
-                }
+            INFO(errors);
+        }
+        REQUIRE(doc->validate());
 
-                mx::InputPtr input = valueElem->asA<mx::Input>();
-                if (input && !input->getConnectedNode())
-                {
-                    std::string interfaceName = input->getNamePath();
-                    interfaceName = nodeDef->createValidChildName(interfaceName);
-                    newGraph->addInterfaceName(input->getNamePath(newGraph), interfaceName);
-                    REQUIRE(nodeDef->getChild(interfaceName));
-                    try
-                    {
-                        // Check duplicate failure case
-                        newGraph->addInterfaceName(input->getNamePath(newGraph), interfaceName);
-                    }
-                    catch (mx::Exception& e)
-                    {
-                        REQUIRE(e.what());
-                        newGraph->removeInterfaceName(input->getNamePath(newGraph));
-                        REQUIRE(nodeDef->getChild(interfaceName) == nullptr);
-                        newGraph->addInterfaceName(input->getNamePath(newGraph), interfaceName);
-
-                        const std::string newInterfaceName = interfaceName + "_renamed";
-                        newGraph->modifyInterfaceName(input->getNamePath(newGraph), newInterfaceName);
-                        REQUIRE(nodeDef->getChild(newInterfaceName));
-                    }
-                }
-            }
+        // Try and fail to create the same definition
+        mx::NodeDefPtr temp;
+        try
+        {
+            temp = nullptr;
+            temp = doc->addNodeDefFromGraph(graph, newNodeDefName, NODENAME, newGraphName);
+        }
+        catch (mx::Exception&)
+        {
+            REQUIRE(temp == nullptr);
         }
 
         // Add new version 
         const std::string VERSION2 = "2.0";
         newGraphName = mx::EMPTY_STRING;
-        nodeDef = doc->addNodeDefFromGraph(graph, newNodeDefName + "2", NODENAME, VERSION2, isDefaultVersion, GROUP, newGraphName);
+        newNodeDefName = doc->createValidChildName("ND_" + graph->getName() + "_2");
+        newGraphName = doc->createValidChildName("NG_" + graph->getName() + "_2");
+        nodeDef = doc->addNodeDefFromGraph(graph, newNodeDefName + "2", NODENAME, newGraphName);
+        nodeDef->setVersionString(VERSION2);
+        //nodeDef->setDefaultVersion(isDefaultVersion);
```

Comment:
Should this commented line of code now be removed?

Reply:
I've removed this. Also I put in node instance creation and proper default version checking when the default changes.